### ISSUE TITLE
If you continue to browse this website... does not allow the services

### ISSUE THREAD

**Issue by alanmac** [AmauriC/tarteaucitron.js#191]
Hello there,

I'm having an issue with the implicit acceptance on navigation. The implication is that by continuing to browse the website (or scroll greater than twice the height of the screen) that tarteaucitron will allow the configured third-party services.

I find that the scroll trigger works, but continuing to browse the site does not until the user clicks "OK, accept all".

My expectation is (and as the language used suggests) that by browsing to another page the user is accepting the third-party services.

I have highPrivacy set to false.

I'm hoping someone can give me some guidance?

Many thanks.

Alan

**Comment by d4rk694** [comment]
I had a similar problem, but if the user clicks ok "Ok" the next time it was asked again to accept the cookies.
But the problem was that I had two cookies tarteaucitron one "mydomaine.com" and the other one ".mydomain.com" and one had the values set as "wait" (I don't remember which one) but one was set by clicking on "OK" or scrolling and the other one by coming again on the website.

I removed all the cookie, and it solved the problem.
Otherwise can you give me the configuration you use and the manipulation you have made to have this behavior?

**Comment by alanmac** [comment]
Hi,

Thanks for your repsonse. I have a fairly default configuration:
```
        tarteaucitron.init({
            "hashtag": "#tarteaucitron", /* Automatically open the panel with the hashtag */
            "highPrivacy": false, /* disabling the auto consent feature on navigation? */
            "orientation": "bottom", /* the big banner should be on 'top' or 'bottom'? */
            "adblocker": false, /* Display a message if an adblocker is detected */
            "showAlertSmall": true, /* show the small banner on bottom right? */
            "cookieslist": true, /* Display the list of cookies installed ? */
            "removeCredit": true, /* remove the credit link? */
            "handleBrowserDNTRequest": true, /* Deny everything if DNT is on */
            //"cookieDomain": ".example.com" /* Domain name on which the cookie for the subdomains will be placed */
        });
```

I have Google Tag Manager set up as third party service. The only tag that it is currently responsible for firing is the Google Universal Analytics tag.

When I first visit the website, the cookie notice is displayed. If I either scroll down past the trigger point or click the OK button this loads the Google Tag Manager and the Google Analytics Tag fires and records my activity on the site. This is all as expected.

What does not work as expected is that if I do not scroll past the trigger point or click the OK buttton, but instead choose to navigate through the site (e.g. clicking the navigation buttons), this navigation does not load the services (i.e. load Tag Manager, and fire the Analytics tag). I would expect that if a user visits the site and then proceeds to navigate (without scrolling or clicking OK to the cookie banner), that tarteaucitron would then register this as an acceptance and load the services... but this does not happen.

I would also hope that before the navigation happens that it would trigger the services that would have loaded on the first page that the user visited, because otherwise the visit to the first page (including referrer) would not get registered with Google Analytics.

**Comment by alanmac** [comment]
Thanks for testing @d4rk694 . I believe this may have been an issue while testing Tag Manager in Debug Mode. It is not happening for me now in production. It is working as expected - when I navigate away from the first page, the next page does not display the banner and the services are loaded - good news!

To your point regarding never triggering services for the first page - would you not consider it acceptable to load the services for the users landing page if the user has started to navigate away from that page having read the banner that if they "continue to browse this website, you are allowing all third-party services". If for example, we don't have any metrics for the homepage visits sent to Google Analytics because it is the most visited entry page to the site, and the services only load from the second page onwards (assuming the user hasn't scrolled a sufficient amount or clicked OK on the banner to activate them on that first page), we would be missing out on some important metrics even though that  user has signaled their agreement to giving by continuing to use the site.

What I am wondering is - is it an option to intercept a navigation link that the user clicks from the first page, load the services and then navigate to the second page. Perhaps this behaviour could be a configuration setting if some people may not want it as the default.

From a GDPR point of view, if a user has signalled their acceptance to using third parties (and that acceptance is the affirmative action of them starting to navigate the site) then I believe it is acceptable to initiate the services for the first page that they are currently on as soon as they begin to navigate but not necessarily have to wait until the second page is loaded to begin being able to report metrics to services like GA.

I'd be interested in your thoughts.

Thanks.

**Comment by d4rk694** [comment]
Your welcome
I'm happy to hear that there is no more problems.

I have no idea if it's possible or if the GDPR allows this behaviour, I'm not an expert in this field..
Maybe you can create a new issue to ask the feature or ask if the GDPR allows it.

**Comment by alanmac** [comment]
Thanks @d4rk694. I'll log the initing on the entry page as a separate issue to discuss.
